crates/release_plz/tests/all/release.rs

```diff
@@ -128,6 +129,12 @@ async fn release_plz_releases_a_new_project_with_custom_release() {
     "#;
     context.write_release_plz_toml(config);
 
+    // Test publish = true in release-plz config but publish = false in manifest
+    let cargo_toml_path = context.repo_dir().join("Cargo.toml");
+    let mut cargo_toml = LocalManifest::try_new(&cargo_toml_path).unwrap();
+    cargo_toml.data["package"]["publish"] = false.into();
+    cargo_toml.write().unwrap();
```

Comment:
could we also test that no packages are published to the registry in this case?

You can download packages from gitea like this:
https://github.com/eviltak/release-plz/blob/843c55c2781bf460a0f1d6e10de08bf57bdbc19c/crates/release_plz/tests/all/changelog.rs#L70